syntax = "proto3";
import "common.proto";
import "google/protobuf/empty.proto";
option go_package = ".;proto";

service Message {
  rpc GetMessagePageList (MessageFilterPageRequest) returns (MessagePageResponse); // 批量获取留言信息
  rpc CreateMessage (MessageRequest) returns (MessageResponse); // 添加留言
  rpc DeleteMessageByIds(common.IdsRequest) returns(google.protobuf.Empty);
}

message MessageFilterPageRequest {
  common.PageRequest pageRequest = 1;
  int32 userId = 2;
  int32 messageType = 3;
  string subject = 4;
  string message = 5;
  string file = 6;
}

message MessageRequest {
  uint64 id = 1;
  int32 userId = 2;
  int32 messageType = 3;
  string subject = 4;
  string message = 5;
  string file = 6;
}

message MessagePageResponse {
  common.PageResponse page = 1;
  repeated MessageResponse list = 2;
}
message MessageResponse {
  uint64 id = 1;
  int32 userId = 2;
  int32 messageType = 3;
  string subject = 4;
  string message = 5;
  string file = 6;
}
